src/Generator/SchemaMapperGenerator.php

```diff
@@ -273,9 +273,16 @@ protected function generateMapStatementsForObject(Field $root, Variable $payload
                     'toSchema',
                     [$requiredResponseItems[$i]]
                 );
-            } elseif ($field->isDate()) {
+            } elseif ($field->isDate() && !$field->isNullable()) {
                 $this->addImport(DateTimeImmutable::class);
                 $requiredVars[] = $this->builder->new('DateTimeImmutable', [$requiredResponseItems[$i]]);
+            } elseif ($field->isDate() && $field->isNullable()) {
```

Comment:
Lets move $field->isDate() above not to duplicate the condition.

Reply:
fixed, please check